### PR TITLE
Add custom annotations for message

### PR DESCRIPTION
**Pull request by lkmtue**
This pull request makes it possible to add custom annotations to compiled messages.

Example:

```
message FooMessage {
  option (scalapb.message).annotations = "@CustomAnnotation";
  ...
}
```
will compile to
```
...
@SerialVersionUID(0L)
@CustomAnnotation
final case class FooMessage(
...
```

### CODE REVIEW

File: compiler-plugin/src/main/scala/com/trueaccord/scalapb/compiler/ProtobufGenerator.scala [hunk 1]
```diff
@@ -1316,6 +1319,10 @@ class ProtobufGenerator(val params: GeneratorParams) extends DescriptorPimps {
 
     serviceFiles ++ enumFiles ++ messageFiles :+ fileDescriptorObjectFile
   }
+
+  private def getAnnotationList(message: Descriptor) = {
+    message.getOptions.getExtension[MessageOptions](Scalapb.message).getAnnotationsList().asScala.toSeq
```
Comment (thesamet) [inline compiler-plugin/src/main/scala/com/trueaccord/scalapb/compiler/ProtobufGenerator.scala]: For consistency with all other options: please move this function to `MessageDescriptorPimps` in `DescriptorPimps.scala`. Rename it to `annotationsOption` and use the `scalaOptions` member to get the extension.

File: protobuf/scalapb/scalapb.proto [hunk 2]
```diff
@@ -61,6 +61,8 @@ message MessageOptions {
 
   // Additional classes and traits to mix in to the companion object.
   repeated string companion_extends = 2;
+
+  repeated string annotations = 3;
```
Comment (thesamet) [inline protobuf/scalapb/scalapb.proto]: Add `// Custom annotations to add to the generated case class.`

File: e2e/src/main/protobuf/custom_annotation.proto [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+package com.trueaccord.proto.e2e;
+
+import "scalapb/scalapb.proto";
+
+option (scalapb.options) = {
+  import: "com.trueaccord.pb.CustomAnnotation"
+};
+
+message CustomAnnotationMessage {
+  option (scalapb.message).annotations = "@CustomAnnotation";
```
Comment (thesamet) [inline e2e/src/main/protobuf/custom_annotation.proto]: Please add another test with 3 different annotations, to verify that the generated code works in that case too.

**Comment by lkmtue** [comment]
Thanks @thesamet for your review. Your comments have been resolved.

**Pull request merged**